crypto/asn1/asn1_parse.c

```diff
@@ -28,6 +28,8 @@ static int asn1_print_info(BIO *bp, long offset, int depth, int hl, long len,
     long saved_indent = -1;
     int i = 0;
 
+    if (bp == NULL)
```

Comment:
With the change of the fuzz driver to use NULL bio, this function is no longer fuzzed. Which does not necessarily mean this change is wrong, but that the fuzz driver change is.

Reply:
I think it should be okay this way since the filename `fuzz/asn1parse.c` and the comment
```
 * Fuzz the parser used for dumping ASN.1 using "openssl asn1parse".
```
suggests that the aim of the fuzzing is the parsing, not the output.
BTW, unfortunately `ASN1_parse()` and friends are not documented.

Yet in order to be more on the conservative side I moved the exclusion of `bp == NULL` to where it is strictly needed.

Reply:
As I said, I'd prefer keeping it like this as there is no point in doing all the stuff in the function does not make any sense with NULL bp and just change the driver back.

Reply:
As agreed below, now a non-NULL mem BIO is used, such that all of the output code is part of the fuzzing as well (as a by-product, though not clearly documented/required).

Nevertheless IMHO it would be pretty nice to allow for NULL `bp` input to `ASN_parse()` and friends, which 
* only needs one `bp == NULL` check and 
* can be very helpful for applications interested only in parsing but not in the diagnostic output it can provide.